src/prefect/tasks/filehandling/compression.py

```diff
@@ -0,0 +1,249 @@
+                like the zip archive (without extension). If True, the basename of the zip archive will
+                be attached to the target_directory and may create a new folder. (default: False)
+            - remove_after_unzip (bool, optional): Removes the zip archive after
+                extraction (default: False)
+
+        Returns:
+            - Target directory as a Path object
+
+        Raises:
+            - ValueError: if source_file not found
+            - TypeError: if source_file is not a zip file
+            - ValueError: if target_directory not found and create_target_if_not_exists = False
+        """
+        zip_file_path = Path(zip_file_path)
+        target_directory = Path(target_directory)
+        if not zip_file_path.is_file():
+            raise ValueError(f"Source file ({zip_file_path}) not found!")
+        elif not zipfile.is_zipfile(zip_file_path):
+            raise TypeError(f"Source file ({zip_file_path}) is not a zip file")
+
+        self._check_target_path(target_directory, create_target_if_not_exists)
+
+        if use_filename_as_target_dir:
+            target_directory = target_directory.joinpath(zip_file_path.stem)
+            target_directory.mkdir(exist_ok=True)
+
+        self.logger.info(f"Extracting {zip_file_path} to {target_directory}")
+
+        if zip_password:
+            zip_password = bytes(zip_password, "utf-8")
+        with ZipFile(zip_file_path, "r") as zip:
+            zip.extractall(target_directory, pwd=zip_password)
+
+        if remove_after_unzip:
+            self.logger.info(f"Removing {zip_file_path} after extraction.")
+            os.remove(zip_file_path)
+
+        return target_directory
+
+
+class Zip(FileBase):
+    """
+    Task to create a zip archive.
+
+    Args:
+        - source_path (Union[str, Path, List[str], List[Path]], optional): a single path or a list of
+            multiple paths for compression into a single zip archive.
+        - target_directory (Union[str, Path], optional): path to the target directory
+        - zip_file_name (str, optional): name of the zip archive. If not set the name will be generated
+            by the following rules:
+                1. if the first object in the source_path is a single file
+                    => basename of the source file without extension
+                2. if the first object in the source_path is a directory
+                    => parent name of the directory
+        - create_target_if_not_exists (bool, optional): Create the target directory if it
+            doesn't exist (default: False)
+        - compression: (int, optional): pass in a supported compression method
+            https://docs.python.org/3/library/zipfile.html?highlight=zipfile#zipfile-objects
+        - compression_level(int, optional): pass in a supported compression level
+            https://docs.python.org/3/library/zipfile.html?highlight=zipfile#zipfile-objects
+            Only supported on Python >= 3.7
```

Comment:
Same here. This is a lot of options, some of which I find confusing (`zip_file_name` and `target_directory` in particular). I'd write this as (in both the constructor and `run` method):

```python
"""
- source (Union[str, Path, List[str], List[Path]], optional): a path or paths to compress
     into a single zip archive.
- target (Union[str, Path], optional): path to the output archive file. Any parent directories
    of `target` must already exist.
- compression_method (str, optional): the compression method to use. Options are
    "deflate", "store", "bzip2", and "lzma". Defaults to `"deflate"`.
- compression_level (int, optional). Compression level to use,
    see https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile for more info.
"""
```

A few notes:
- We don't support inferring the directory name from the inputs. Inference is unlikely to satisfy most users, and doesn't really make sense for archives created from a list of paths. If a user wants to determine the archive path from the input, they can write a short `@task` decorated task to do this themselves.
- All directories must already exist. Again, an `@task` decorated task can do this external to this task if needed.
- We use string codes instead of integers for `compression_method`. This is much friendlier than the stdlib int-based interface.
- We default to `deflate` instead of `store`. This better matches the existing `zip` shell function (which compresses by default).